pkg/config/config.go

```diff
@@ -23,70 +25,151 @@ import (
-	gracePeriodConfigKey                    = "GRACE_PERIOD"
-	podTerminationGracePeriodConfigKey      = "POD_TERMINATION_GRACE_PERIOD"
-	podTerminationGracePeriodDefault        = -1
+	nodeNameConfigKey                       = "NODE_NAME"
 	nodeTerminationGracePeriodConfigKey     = "NODE_TERMINATION_GRACE_PERIOD"
 	nodeTerminationGracePeriodDefault       = 120
+	podTerminationGracePeriodConfigKey      = "POD_TERMINATION_GRACE_PERIOD"
+	podTerminationGracePeriodDefault        = -1
 	webhookURLConfigKey                     = "WEBHOOK_URL"
 	webhookURLDefault                       = ""
 	webhookHeadersConfigKey                 = "WEBHOOK_HEADERS"
 	webhookHeadersDefault                   = `{"Content-type":"application/json"}`
 	webhookTemplateConfigKey                = "WEBHOOK_TEMPLATE"
 	webhookTemplateDefault                  = `{"text":"[NTH][Instance Interruption] EventID: {{ .EventID }} - Kind: {{ .Kind }} - Description: {{ .Description }} - State: {{ .State }} - Start Time: {{ .StartTime }}"}`
-	enableScheduledEventDrainingConfigKey   = "ENABLE_SCHEDULED_EVENT_DRAINING"
-	enableScheduledEventDrainingDefault     = false
-	enableSpotInterruptionDrainingConfigKey = "ENABLE_SPOT_INTERRUPTION_DRAINING"
-	enableSpotInterruptionDrainingDefault   = true
 )
 
 //Config arguments set via CLI, environment variables, or defaults
 type Config struct {
-	DryRun                         bool
-	NodeName                       string
-	MetadataURL                    string
-	IgnoreDaemonSets               bool
-	DeleteLocalData                bool
-	KubernetesServiceHost          string
-	KubernetesServicePort          string
-	PodTerminationGracePeriod      int
-	NodeTerminationGracePeriod     int
-	WebhookURL                     string
-	WebhookHeaders                 string
-	WebhookTemplate                string
-	EnableScheduledEventDraining   bool
-	EnableSpotInterruptionDraining bool
+	DeleteLocalData                bool   `json:"delete-local-data"`
+	DryRun                         bool   `json:"dry-run"`
+	EnableScheduledEventDraining   bool   `json:"enable-scheduled-event-draining"`
+	EnableSpotInterruptionDraining bool   `json:"enable-spot-interruption-draining"`
+	IgnoreDaemonSets               bool   `json:"ignore-daemon-sets"`
+	KubernetesServiceHost          string `json:"kubernetes-service-host"`
+	KubernetesServicePort          string `json:"kubernetes-service-port"`
+	MetadataURL                    string `json:"metadata-url"`
+	NodeName                       string `json:"node-name"`
+	NodeTerminationGracePeriod     int    `json:"node-termination-grace-period"`
+	PodTerminationGracePeriod      int    `json:"pod-termination-grace-period"`
+	WebhookURL                     string `json:"webhook-url"`
+	WebhookHeaders                 string `json:"webhook-headers"`
+	WebhookTemplate                string `json:"webhook-template"`
+}
+
+var flagData = map[string]map[string]interface{}{
+	"delete-local-data": map[string]interface{}{
```

Comment:
instead  of defining an interface{} for each of these, could we pull that out to a struct with key, defValue, and usage as fields? defValue would still be of interface{} type but key and usage could be typed to string and then you wouldn't have to hardcode the fields when you're looking things up below.